algorithms/linfa-ica/benches/fast_ica.rs

```diff
@@ -0,0 +1,97 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use linfa::{dataset::DatasetBase, traits::Fit};
+use linfa_ica::fast_ica::{FastIca, GFunc};
+use ndarray::{array, concatenate};
+use ndarray::{Array, Array2, Axis};
+use ndarray_rand::{rand::SeedableRng, rand_distr::Uniform, RandomExt};
+use rand_xoshiro::Xoshiro256Plus;
+
+fn perform_ica_cube(size: usize) {
+    let sources_mixed = create_data(size);
+
+    let ica = FastIca::params().gfunc(GFunc::Cube).random_state(10);
+
+    let ica = ica.fit(&DatasetBase::from(sources_mixed.view()));
+}
+
+fn perform_ica_exp(size: usize) {
+    let sources_mixed = create_data(size);
+
+    let ica = FastIca::params().gfunc(GFunc::Exp).random_state(10);
+
+    let ica = ica.fit(&DatasetBase::from(sources_mixed.view()));
+}
+
+fn perform_ica_logcosh(size: usize) {
+    let sources_mixed = create_data(size);
+
+    let ica = FastIca::params()
+        .gfunc(GFunc::Logcosh(1.0))
+        .random_state(10);
+
+    let ica = ica.fit(&DatasetBase::from(sources_mixed.view()));
+}
```

Comment:
These 3 can be refactored into a single function `perform_ica(size: usize, gfunc: GFunc)`